extras/test_hull_performace.cpp

```diff
@@ -0,0 +1,175 @@
+
+#include <CGAL/Exact_predicates_exact_constructions_kernel.h>
+#include <CGAL/Exact_predicates_inexact_constructions_kernel.h>
+#include <CGAL/Polygon_mesh_processing/corefinement.h>
+#include <CGAL/Polygon_mesh_processing/measure.h>
+#include <CGAL/Surface_mesh.h>
+#include <CGAL/convex_hull_3.h>
+
+#include <algorithm>
+#include <chrono>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <random>
+#include <sstream>  // For string manipulation
+
+#include "manifold.h"
+#include "meshIO.h"
+#include "samples.h"
+using namespace std;
+using namespace glm;
+using namespace manifold;
+
+// Epick = Exact predicates Inexact constructions. Seems fair to use to compare
+// to Manifold, which seems to store double coordinates.
+typedef CGAL::Epick Kernel;
+
+// Epeck = Exact predicates Exact constructions. What OpenSCAD uses to guarantee
+// geometry ends up where it should even after many operations. typedef
+// CGAL::Epeck Kernel;
+
+typedef CGAL::Point_3<Kernel> Point;
+typedef CGAL::Surface_mesh<Point> TriangleMesh;
+typedef CGAL::SM_Vertex_index Vertex;
+
+// Converts Manfiold to CGAL Surface Mesh
+void manifoldToCGALSurfaceMesh(Manifold &manifold, TriangleMesh &cgalMesh) {
+  auto maniMesh = manifold.GetMesh();
+
+  const size_t n = maniMesh.vertPos.size();
+  std::vector<Vertex> vertices(n);
+  for (size_t i = 0; i < n; i++) {
+    auto &vert = maniMesh.vertPos[i];
+    vertices[i] = cgalMesh.add_vertex(Point(vert.x, vert.y, vert.z));
+  }
+
+  for (auto &triVert : maniMesh.triVerts) {
+    std::vector<Vertex> polygon{vertices[triVert[0]], vertices[triVert[1]],
+                                vertices[triVert[2]]};
+    cgalMesh.add_face(polygon);
+  }
+}
+
+// Prints the volume and area of the manifold and the convex hull of our
+// implementation,
+void PrintVolArea(Manifold &manfiold_obj, Manifold &hullManifold) {
+  std::cout << manfiold_obj.GetProperties().volume << ",";
+  std::cout << hullManifold.GetProperties().volume << ",";
+  std::cout << manfiold_obj.GetProperties().surfaceArea << ",";
+  std::cout << hullManifold.GetProperties().surfaceArea << ",";
+  std::cout << manfiold_obj.NumTri() << ",";
+  std::cout << hullManifold.NumTri() << ",";
+  return;
+}
+
+// Prints the volume and area of the manifold and Convex Hull of CGAL
+// Implementation
+void PrintVolAreaCGAL(TriangleMesh &cgalMesh, TriangleMesh &cgalHull) {
+  std::cout << CGAL::Polygon_mesh_processing::volume(cgalMesh) << ",";
+  std::cout << CGAL::Polygon_mesh_processing::volume(cgalHull) << ",";
+  std::cout << CGAL::Polygon_mesh_processing::area(cgalMesh) << ",";
+  std::cout << CGAL::Polygon_mesh_processing::area(cgalHull) << ",";
+  std::cout << std::distance(cgalMesh.faces_begin(), cgalMesh.faces_end())
+            << ",";
+  std::cout << std::distance(cgalHull.faces_begin(), cgalHull.faces_end())
+            << ",";
+  return;
+}
+
+void PrintManifold(Manifold input, Manifold (Manifold::*hull_func)() const) {
```

Comment:
Why should this take a manifold method? I think you can just make it take a generic function that returns a manifold given a set of points.